### ISSUE TITLE
Reliable URI-URL for accessibility standards conformance

### ISSUE THREAD

**Issue by LaZay** [mattgarrish/epub-revision#2408]
In the v1.0 specification, the accessibility standards conformance of a publication must be addressed only by a formal and reliable URI-URL:
- [http://www.idpf.org/epub/a11y/accessibility-20170105.html#wcag-a](http://www.idpf.org/epub/a11y/accessibility-20170105.html#wcag-a)
- [http://www.idpf.org/epub/a11y/accessibility-20170105.html#wcag-aa](http://www.idpf.org/epub/a11y/accessibility-20170105.html#wcag-aa)
- [http://www.idpf.org/epub/a11y/accessibility-20170105.html#wcag-aaaa](http://www.idpf.org/epub/a11y/accessibility-20170105.html#wcag-aaa)

These URL were fully documented in the [functional ](https://www.w3.org/Submission/epub-a11y/#sec-conf-reporting) spec - not in the [technical specification ](https://idpf.org/epub/a11y/techniques/techniques.html#sec-discovery) though. But that is another issue.

These URLs provided only 2 types of information:
1. Version of A11Y EPUB spec
2. WCAG level.

In version 1.1 of the spec, the accessibility standards conformance must now provide 3 types of information:
1. A11Y EPUB spec version
2. WCAG version
3. WCAG level.

This is much better due to the important changes between versions 2.0, 2.1, and 3.0 of the WCAG.

The [functional spec ](https://www.w3.org/TR/epub-a11y-11/#sec-conf-reporting-pub) - still not the [technical one](https://www.w3.org/TR/epub-a11y-tech-11/#sec-discovery)  - says that these information must now be provided by a string, no more via URLs ([missing](https://www.w3.org/TR/WCAG/#conformance-required)).

The string is fully documented, and formalized by a string pattern (upcoming problem with "-" separator probably). Hoping this will be controlled by epubcheck.
No more URL is provided/nor documented by the spec. A note says that the WCAG should provide these URLs, but did not yet.

But what should the WCAG provide exactly? What URL for what conformance? Maybe it is time for stoping concatening different types information under the same static unique URL. All the problems come from here as a matter of fact. Why not manage a dynamic combination of well separated accessibility standards information?

**URL to the WCAG version?** 
It already exists. Ex: [https://www.w3.org/WAI/WCAG21](https://www.w3.org/WAI/WCAG21). 
It could be simply set this way in the EPUB OPF metadata:
`<link rel="dcterms:conformsTo" href="https://www.w3.org/TR/WCAG21"/>`

**URL to the WCAG level?** 
That is missing indeed. Very easy to solve shortly though, by simply adding anchors [there](https://www.w3.org/TR/WCAG21/#cc1) on A / AA / AAA list items.
It could be simply set this way in the very near future:
```
<link rel="dcterms:conformsTo" href="https://www.w3.org/TR/WCAG21/#cc1A"/>
<link rel="dcterms:conformsTo" href="https://www.w3.org/TR/WCAG21/#cc1AA"/>
<link rel="dcterms:conformsTo" href="https://www.w3.org/TR/WCAG21/#cc1AAA"/>
```

**URL to the A11Y EPUB spec version?**
The WCAG is not concerned I am afraid. The URL of the v1.1 EPUB A11y spec should be fine.
It could be simply set this way:
`<link rel="dcterms:conformsTo" href="https://www.w3.org/TR/epub-a11y-11"/>`

And as `dcterms:conformsTo` is not accurate enough, another attribute could be added in the future to specifiy the type of accessibility standard that is addressed.

**Comment by mattgarrish** [comment]
> A note says that the WCAG should provide these URLs, but did not yet.

WCAG provides a URL you can use in the [linked section](https://www.w3.org/TR/WCAG/#conformance-required) (see the second list item).

The problem, going back to 1.0, was always its lack of expressiveness. We initially were looking at having two conformance claims - one for epub and one for wcag - but in the end the only way to make it all work in a way that wasn't excessively cumbersome for authoring and processing was to merge the information to a single claim of our own making.

We also learned from 1.0 that URLs are not very good for expressing the claims we need. We already [requested anchors](https://github.com/w3c/wcag/issues/1575), but they're not coming until WCAG 2.2 is released, so 2.0 and 2.1 will forever lack them. Anyone trying to resolve a 2.0 or 2.1 URL with non-existent anchors won't reach the definition of what those links refer to.

That's one of the reasons why we've moved away from URLs for 1.1. The other big one was having a URL for an organization that no longer exists, so there was no way to adapt what we had to continue its use moving forward.

We also recently resolved to make the conformance claim as human readable as possible, while also retaining machine readability, to avoid redundancies for users with the summary, so I don't see a return to URLs being realistic now.

That said, if you personally want to put URLs in your EPUBs in addition to the required conformance statement, we're not stopping you with this specification. That's what the note is saying - we don't own the property, so we're not banning other uses of it. But we're also not endorsing or defining other uses.

**Comment by LaZay** [comment]
Thks for your long explaination.

Precisely, when I looked at the [5.3.1 Required Components of a Conformance Claim](https://www.w3.org/TR/WCAG/#conformance-required), I saw that the only URL provided by the WCAG was the one concerning WCAG version. I.e. https://www.w3.org/TR/WCAG21. Which is not enough... :-/

It is usually better doing what everybody does, to facilate the EPUB production, and guarantee robust exploitations of metadata by reading systems, bookseller, libraries...

We will see if there are other reactions on this ticket.

**Comment by mattgarrish** [comment]
Propose that we close this issue, too. We've had a lot of discussion around the identifier already in getting to the strings that we now use, so I can't see reversing the work at this point unless a major problem is identified. /cc @avneeshsingh

**Comment by avneeshsingh** [comment]
As @mattgarrish mentioned, we had long discussions for using URI vs predefined strings, and the group settled on using the predefined strings. We made further decisions for AccessibilitySummary on the basis of predefined strings (#2116. 
Since we are already in the CR stage, I also think that it is too late to reverse all the major changes that we have made on the basis of discussions that we have been having for last two years, until and unless we really hit a blocking issue.
In case WCAG comes up with proper URI for indicating version of WCAG and level A, AA, AAA, then it can be a trigger for revisiting this decision in next versions of EPUB Accessibility.
